### PR TITLE
[TRTLLM-13696][test] Part2.3: Migrate CPU only tests - auto deploy

### PR DESCRIPTION
**Pull request by tongyuantongyu**
## Dev Engineer Review

- Marked non-GPU AutoDeploy tests with `pytest.mark.cpu_only`.
- Added AutoDeploy suites to `tests/integration/test_lists/test-db/l0_cpu.yml`.
- Refactored executor fixtures and configuration test setup.
- Updated quantization coverage for bias and no-bias cases.
- Added a CUDA-availability skip for the FP8 resource-handler test.
- No public API changes were identified.
- CPU-stage CI passed on `CPU-Generic-x86-1` and `CPU-Generic-arm-1`.
- Full CI remained `UNSTABLE`. Multi-GPU tests require the `ci: full pre-merge approved` label and follow-up investigation.

## QA Engineer Review

- Updated AutoDeploy tests for model structure, compilation, transformations, custom operations, MLIR, shims, utilities, and standalone imports.
- Added CPU-only markers to the modified test modules and selected test functions.
- Updated test behavior in `test_quant.py`, `test_create_ad_executor.py`, `test_llm_config.py`, and `test_resource_handlers.py`.
- Added `_torch/auto_deploy`, `auto_deploy/multigpu`, `auto_deploy/singlegpu`, and `auto_deploy/standalone` to `tests/integration/test_lists/test-db/l0_cpu.yml`.
- The modified AutoDeploy test coverage is listed in the CPU test database.
- Verdict: needs follow-up because full CI was `UNSTABLE` and another run was requested after failure investigation.

## Description

Move AutoDeploy test cases don't need GPU to CPU stage.

## Test Coverage

The CPU stage.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

File: tests/unittest/auto_deploy/singlegpu/shim/test_llm_config.py [hunk 1]
```diff
@@ -112,27 +114,28 @@ def test_cache_transceiver_rejects_unmanaged_persistent_caches():
...
+                    "backend": "flashinfer",
+                },
             },
-            "insert_cached_attention": {
-                "stage": "cache_init",
-                "backend": "flashinfer",
-            },
-        },
-    }
+        }
```
[the remark this answers is not in the document]
Comment (tongyuantongyu) [reply tests/unittest/auto_deploy/singlegpu/shim/test_llm_config.py]: This is to fix this call: https://github.com/NVIDIA/TensorRT-LLM/blob/63eb095097067d704f3267fc31ce89b0e9f73221/tensorrt_llm/llmapi/llm.py#L372-L373

And since `llm.py` is doing a `from ... import ...` call we have to patch the imported symbol directly: https://github.com/NVIDIA/TensorRT-LLM/blob/63eb095097067d704f3267fc31ce89b0e9f73221/tensorrt_llm/llmapi/llm.py#L51-L52

And it's legit to use `gpus_per_node` to **overwrite** the number of GPUs available per node, which defaults to the number reported by `get_device_count`.

File: tests/unittest/auto_deploy/singlegpu/utils/test_multi_stream_utils.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 
 from tensorrt_llm._torch.auto_deploy.utils import multi_stream_utils as msu
 
+pytestmark = pytest.mark.cpu_only
```
Comment (brnguyen2) [inline tests/unittest/auto_deploy/singlegpu/utils/test_multi_stream_utils.py]: Module-level `pytestmark` here removes `TestRecordStreamDuringCudaGraphCapture` (line 108) from CI entirely: the CPU stage selects it via `cpu_only` and then its own `@pytest.mark.skipif(not torch.cuda.is_available())` skips it, while every GPU stage deselects it with `-m "not cpu_only"`. That class is the regression guard for the `record_stream`-during-capture bug (silent shared-expert input corruption → runaway generation), so this is exactly the test you least want to lose.

Suggest per-test marks instead — apply `@pytest.mark.cpu_only` to the passthrough tests above and leave the CUDA class unmarked.

File: tests/unittest/auto_deploy/singlegpu/models/test_hf.py [hunk 3]
```diff
@@ -27,6 +27,8 @@
     hf_load_state_dict_with_device,
 )
 
+pytestmark = pytest.mark.cpu_only
```
Comment (brnguyen2) [inline tests/unittest/auto_deploy/singlegpu/models/test_hf.py]: Same pattern, milder: `test_hf_load_state_dict_with_device` has an `if torch.cuda.is_available():` branch (lines 72–82) asserting `device_map={"": "cuda"}`. With the module marked `cpu_only`, that branch runs on no CI stage — the CPU stage takes the false path and GPU stages never select the test. Either mark the CPU-safe tests individually and leave this one unmarked, or split the CUDA assertion into its own test so it stays on a GPU stage.

File: tests/unittest/auto_deploy/singlegpu/shim/test_llm_config.py [hunk 4]
```diff
@@ -112,27 +114,28 @@ def test_cache_transceiver_rejects_unmanaged_persistent_caches():
 # ================================
 
 
-@pytest.fixture
+@pytest.fixture(scope="module")
```
Comment (brnguyen2) [inline tests/unittest/auto_deploy/singlegpu/shim/test_llm_config.py]: `test_config_params` is a data fixture, but it now also holds an active `patch("tensorrt_llm.llmapi.llm.get_device_count")` open for the whole module (scope="module" + yield keeps the patch live until module teardown). Once any test requests it, the patch leaks into every later test in the file regardless of whether they want it — an order-dependent global. Cleaner to keep the fixture returning the dict and put the patch in a narrow `with` inside `test_config_flow`, or in a separate function-scoped fixture.

Also `"gpus_per_node": 1` at line 180 with `world_size=3` and `get_device_count` patched to 3 is belt-and-suspenders — a short comment on which one is actually needed to run GPU-less would help the next reader.

Comment (tongyuantongyu) [reply tests/unittest/auto_deploy/singlegpu/shim/test_llm_config.py]: Extract and narrowed the `device_count` patch to the only test need it: `test_config_flow`.

File: tests/unittest/auto_deploy/singlegpu/shim/test_create_ad_executor.py [hunk 5]
```diff
@@ -26,6 +27,33 @@
...
+    with (
+        patch(
+            "tensorrt_llm._torch.auto_deploy.shim.ad_executor.Distributed.get",
+            return_value=mock_dist,
+        ),
+        patch("tensorrt_llm._torch.auto_deploy.shim.ad_executor.torch.cuda.set_device"),
+        patch("tensorrt_llm._torch.auto_deploy.shim.ad_executor.initialize_or_skip"),
+        patch(
+            "tensorrt_llm._torch.auto_deploy.shim.ad_executor.AutoTuner.get",
+            return_value=mock_autotuner,
```
Comment (brnguyen2) [inline tests/unittest/auto_deploy/singlegpu/shim/test_create_ad_executor.py:48]: Non-blocking: with the module now `cpu_only` and `instantiate_sampler` mocked in an autouse fixture, `create_autodeploy_executor`'s sampler-construction path is exercised by no CI stage — GPU stages deselect the module, and here it returns a `Mock()`. Same for the `Distributed.get` / `initialize_or_skip` port-broadcast path. No test in this file asserts on the sampler, so nothing regresses today, but it's a quiet loss of the incidental "does this actually construct" coverage the GPU stage used to give. Worth a follow-up if any of that is meant to be covered.
